### PR TITLE
 [TA-135]  adding tests

### PR DESCRIPTION
**Pull request by Mromellete**
## Summary by CodeRabbit

- Tests
  - Added comprehensive automated test coverage for administrative and association workflows, including profiles, volunteers, documents, reports, dashboards, missions, notifications, and engagements.
  - Validates success paths, pagination, filtering, and robust error handling for API failures, improving reliability and stability.
  - Ensures consistent handling of time-based metrics and edge cases.
  - No user-facing changes; enhances confidence in future releases.

### CODE REVIEW

**Comment by Mromellete** [comment]
@coderabbitai review

File: __tests__/services/adminService.test.ts [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    });
+
+    test("getDashboardStats: if any subcall fails with AxiosError, it rethrows normalized message", async () => {
+        jest.spyOn(adminService, "getDashboardOverview").mockRejectedValueOnce(
+            makeAxiosError("boom", { detail: "Backend says no" }, 500)
+        );
+
+        await expect(adminService.getDashboardStats()).rejects.toThrow("Backend says no");
+        expect(console.error).toHaveBeenCalled(); // from handleApiError
+    });
```
[the remark this answers is not in the document]
Comment (Mromellete) [reply __tests__/services/adminService.test.ts:361]: ignore this and let me merge

File: __tests__/services/adminService.test.ts [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    const apiMock = api as unknown as ApiMock;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        jest.spyOn(console, "error").mockImplementation(() => {});
+    });
+
+    afterEach(() => {
+        (console.error as jest.Mock).mockRestore();
+    });
```
[the remark this answers is not in the document]
Comment (Mromellete) [reply __tests__/services/adminService.test.ts:53]: ignore this and let me merge
